src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -560,8 +560,6 @@ GenTree* Compiler::getArgForHWIntrinsic(var_types            argType,
 
         arg = impPopStack().val;
         assert(varTypeIsArithmetic(arg->TypeGet()) || ((argType == TYP_BYREF) && arg->TypeIs(TYP_BYREF)));
-
-        assert(genActualType(arg->gtType) == genActualType(argType));
```

Comment:
I think you can replace this by `impCheckImplicitArgumentCoercion`

Reply:
In fact, we should probably `BADCODE` if `impCheckImplicitArgumentCoercion` returns false, in the same way as `impPopCallArgs`

Reply:
Oh, TIL: impCheckImplicitArgumentCoercion exists

Reply:
Thanks, addressed

Reply:
78765